pkg/preparer/orchestrate.go

```diff
@@ -79,6 +79,90 @@ func (p podWorkerID) String() string {
 	return fmt.Sprintf("%s-%s", p.podID.String(), p.podUniqueKey)
 }
 
+func (p *Preparer) InstallWhiteListPods(whiteListPods map[types.PodID]bool, preparerConfig *PreparerConfig) error {
+	intentResults, _, err := p.store.ListPods(consul.INTENT_TREE, p.node)
+	realityResults, _, err := p.store.ListPods(consul.REALITY_TREE, p.node)
+	var intents []consul.ManifestResult
+	if err != nil {
+		p.Logger.WithError(err).Errorln("could not check intent")
+		return err
+	}
+	errorChan := make(chan error)
+	quit := make(chan struct{})
+
+	for _, intentResult := range intentResults {
+		intentSHA, _ := intentResult.Manifest.SHA()
+		if _, ok := whiteListPods[intentResult.Manifest.ID()]; ok {
+			// compare intent/reality for the pod here, to prevent reinstall
```

Comment:
`resolvePair` already does this comparison. Is there any reason to do it twice?

Reply:
I just wanted to maintain a clear logic here, that we just install whitelist pods which haven't installed. Yeah, if we let `resolvePair` to decide, we still need the for loop here to compare the manifest.ID(), and put the reality manifest into the `ManifestPair`.

Reply:
Installing only those pods which have not yet been installed is the typical, expected behavior for the preparer. I believe that redundantly performing this check makes the logic less clear because it begs the question of why the typical preparer behavior is insufficient.

Reply:
They are not the exactly the same logic, I have made the upmost the use existing codes. Well, we can do this in either way, I will modify as you've mentioned.